# Trabajo Práctico 1 - Matrices de Insumo Producto

## Enunciado

### Consigna 1 - 

a) $(I − A)$ es inversible :

Si la matriz (I-A) es inversible, el sistema $(I-A)^{-1}$ tiene una única solución. Esto, en términos de lo que el sistema representa, significa que existe una única producción $\textbf{p}$ que satisface tanto la demanda interna como la externa.

b) $(I − A)$  es inversible y $\textbf{d}$ = 0 :

Que $\textbf{d}$ sea 0 significa que estamos hablando de una economía cerrada. Por lo tanto, toda la producción esta de destinada a satisfacer la demanda interna. Además, la matriz $\\$ $(I-A)$ es inversible, lo que implica que podemos encontrar la producción de equilibrio para afrontar la demanda interna.

Teoria: No se produce nada para que no hay demanda externa y por ende la produccion es igual a 0

c) $(I − A)$ no es inversible :

Analizando unicamente el sistema 2, en este caso se deduce que no existe una solucion $\textbf{p}$ que satisface tanto la demanda interna como la externa.

Sin embargo, si vemos el sistema 1, podemos llegar a la ecuacion:

$$
(I - A) \cdot p = d \quad \text{(1)}
$$

Luego, como suponemos que $(I − A)$ no es inversible obtenemos 2 posibles alternativas: que tenga infinitas soluciones o que no tenga solucion. Por lo que puede llegar a pasar que hayan infinitas producciones que satisfacen tanto la demanda externa como interna.

d) $(I − A)$ no es inversible y $\textbf{d}$ = 0 :

### Consigna 3 -  
Resolver el sistema 2 utilizando inversaLU para los siguientes valores de $\textbf{A}$ y $\textbf{d}$:

$$
\mathbf{A} = 
\begin{pmatrix}
0.3 & 0.0 & 0.1 \\
0.05 & 1.0 & 0.2 \\
0.1 & 0.15 & 0.1
\end{pmatrix},
\quad
\mathbf{d} =
\begin{pmatrix}
100 \\
100 \\
300
\end{pmatrix}
$$


1) ¿Hay una justificación económica de los valores encontrados para $\textbf{p}$?
2) ¿Cual es la característica de la matriz que lleva a esta solución?

In [5]:
from funciones import inversaLU, calcularLU
import numpy as np

A = np.array([[0.3, 0.0, 0.1], [0.05, 1.0, 0.2], [0.1, 0.15, 0.1]])

d = [[100], [100], [300]]

I = np.eye(3)

L = inversaLU(I-A)

p = L@d

print(L)

[[  1.37931034  -0.68965517   0.        ]
 [ -2.98850575 -28.50574713  -6.66666667]
 [ -0.34482759  -4.82758621   0.        ]]


1) Sí hay una justificación económica para este resultado. El producto 2 requiere en el segundo sector de la economía una unidad de si mismo para ser producido. Por un lado, esto indica que no es una economía viable ya que nunca se va a poder satisfacer la demanda de producto 2, debido a que se entra en un bucle infinito. Por lo tanto la producción de producto 2 nos da un valor de alrededor de 5000 en modulo porque es una demanda que no esta siendo satisfecha. Por otro lado, notemos que la demanda total de producto 2 es de 1.15 unidades sumando la demanda de los sectores 2 y 3, pero recordemos que la producción de producto 2 necesita una unidad de si mismo. Entonces si restamos la producción de producto 2 y su demanda obtenemos un resultado con signo negativo. Esto es indicador de que nunca vamos a poder producir lo necesario para satisfacer la demanda total para este producto. $\\$ En resumen, las desviaciones observadas en los sectores 2 y 3 son producto de demandar en cantidades que no pueden ser satisfechas al producto 2. Esto derivó en una economía inviable. Los altos valores en modulo indican la necesidad producir más para intentar satisfacer su demanda de producto 2 y el simbolo menos nos indica que a pesar de producir lo máximo nunca se llego a satisfacer esa demanda.

2)  Para calcular el determinante de una matriz 3×3, se puede usar la siguiente fórmula general: $\\$ Para una matriz de la forma:
    $$
    \mathbf{A} = 
    \begin{pmatrix}
    a & b & c \\
    d & e & f \\
    g & h & i
    \end{pmatrix}
    $$

    El determinante se calcula como: det=a(ei−fh)−b(di−fg)+c(dh−eg)

    En este caso. la matriz es:
    $$
    \mathbf{A} = 
    \begin{pmatrix}
    1.37931034 & -0.68965517 & 0 \\
    -2.98850575 & -28.50574713 & -6.66666667 \\
    -0.34482759 & -4.82758621 & 0
    \end{pmatrix}
    $$

    Sustituyendo la fórmula: $\\$
    det= 1.37931034⋅(−32.1848725)−((−0.68965517)⋅(−2.2955691)) = −46.0115


### Consigna 4 -  
Analizar analíticamente qué sucede con la producción total cuando a partir de una demanda externa $\textbf{d}$, la demanda externa para el producto 3 cambia en un $\textbf{∆d}$. $\\$ La demanda externa se convierte en:
    $$
    \mathbf{d'} = \mathbf{d}+
    \begin{pmatrix}
    0 \\
    0 \\
    1
    \end{pmatrix}
    $$

Conocemos la economía con $\textbf{d}$, que esta modelada por:
$$\mathbf{p}= \mathbf{Ld}$$
Ahora calculemos la economía con $\textbf{d'}$

$$\mathbf{p'} = \mathbf{Ld'} = \mathbf{L(d+∆d)}$$

$$\mathbf{p + ∆p} = \mathbf{Ld} + \mathbf{L∆d}$$

Sabiendo que $\textbf{p = Ld}$ cancelamos y obtenemos que:

$$\mathbf{∆p} = \mathbf{L∆d} \\$$
Reemplazando :
$$
\mathbf{∆d} = \begin{pmatrix}
    0 \\
    0 \\
    1
    \end{pmatrix}
$$

y obtenemos

$$\mathbf{∆p} = \mathbf{L} \begin{pmatrix}
    0 \\
    0 \\
    1
    \end{pmatrix}\\$$

Lo cual implica un aumento en la demanda del producto 3.
